src/main/java/health/ere/ps/websocket/Websocket.java

```diff
@@ -95,7 +97,10 @@ public void onMessage(String message) {
 
                 SignAndUploadBundlesEvent event = new SignAndUploadBundlesEvent(object);
                 signAndUploadBundlesEvent.fireAsync(event);
-            }
+            } else if("XMLBundle".equals(object.getString("type"))) {
+                Bundle[] bundles = XmlPrescriptionProcessor.parseFromString(object.getString("payload"));
+                onFhirBundle(new BundlesEvent(bundles));
```

Comment:
Nit: Why not -  BundlesEvent event = new BundlesEvent(bundles); 

bundlesEvent.fireAsync(event); ?